### ISSUE TITLE
LXC 1.0.8 lxc.cgroup.devices.allow Error: safe_mount: 1284 No such file or directory

### ISSUE THREAD

**Issue by naisanza** [lxc/lxc#994]
Migrated from https://github.com/lxc/lxd/issues/1976
# Required information
- Distribution: Ubuntu 
- Distribution version: 14.04.4 LTS
- The output of "lxc info" or if that fails:
  - Kernel version: 3.13.0-83-generic
  - LXC version: 1.0.8-0ubuntu0.3
  - LXD version: n/a
  - Storage backend in use: ext4
# Issue description

When configured to use `lxc.cgroup.devices.allow = c 4:9 rw` the container is not able to start. And I'm not sure what the error message means, or what I need to resolve the missing directory

Error in question:

```
lxc_container: utils.c: safe_mount: 1284 No such file or directory - Mount of '/sys/firmware/efi/efivars' onto '/usr/lib/x86_64-linux-gnu/lxc/sys/firmware/efi/efivars' failed
```

When trying to create the directory, it shows the error:

```
eric@Shared:~$ sudo mkdir -p /sys/firmware/efi/efivars
mkdir: cannot create directory ‘/sys/firmware/efi’: No such file or directory
eric@Shared:~$ cd /sys/firmware/
eric@Shared:/sys/firmware$ lsattr
lsattr: Inappropriate ioctl for device While reading flags on ./acpi
lsattr: Inappropriate ioctl for device While reading flags on ./memmap
eric@Shared:/sys/firmware$ ls -hl
total 0
drwxr-xr-x  5 root root 0 Apr 30 03:57 acpi
drwxr-xr-x 10 root root 0 Apr 30 03:57 memmap
```

lxc-start -n X11:

```
lxc-start -n X11
lxc_container: utils.c: safe_mount: 1284 No such file or directory - Mount of '/sys/firmware/efi/efivars' onto '/usr/lib/x86_64-linux-gnu/lxc/sys/firmware/efi/efivars' failed
lxc_container: lxc_start.c: main: 341 The container failed to start.
lxc_container: lxc_start.c: main: 345 Additional information can be obtained by setting the --logfile and --logpriority options.
```

container config;

```
lxc.mount.entry = proc proc proc nodev,noexec,nosuid 0 0
lxc.mount.entry = sysfs sys sysfs defaults 0 0
lxc.mount.entry = /sys/fs/fuse/connections sys/fs/fuse/connections none bind,optional 0 0
lxc.mount.entry = /sys/kernel/debug sys/kernel/debug none bind,optional 0 0
lxc.mount.entry = /sys/kernel/security sys/kernel/security none bind,optional 0 0
lxc.mount.entry = /sys/fs/pstore sys/fs/pstore none bind,optional 0 0
lxc.mount.entry = /dev/console dev/console none bind,create=file 0 0
lxc.mount.entry = /dev/full dev/full none bind,create=file 0 0
lxc.mount.entry = /dev/null dev/null none bind,create=file 0 0
lxc.mount.entry = /dev/random dev/random none bind,create=file 0 0
lxc.mount.entry = /dev/tty dev/tty none bind,create=file 0 0
lxc.mount.entry = /dev/urandom dev/urandom none bind,create=file 0 0
lxc.mount.entry = /dev/zero dev/zero none bind,create=file 0 0
lxc.mount.entry = /sys/firmware/efi/efivars sys/firmware/efi/efivars none bind,optional 0 0
lxc.mount.entry = /proc/sys/fs/binfmt_misc proc/sys/fs/binfmt_misc none bind,optional 0 0
lxc.tty = 4
lxc.pts = 1024
lxc.arch = x86_64
lxc.seccomp = /usr/share/lxc/config/common.seccomp
lxc.utsname = X11
lxc.network.type = veth
lxc.network.flags = up
lxc.network.link = lxcbr0
lxc.network.hwaddr = 00:16:3e:d7:c7:bc
lxc.cap.drop = sys_module
lxc.cap.drop = mac_admin
lxc.cap.drop = mac_override
lxc.cap.drop = sys_time
lxc.id_map = u 0 100000 65536
lxc.id_map = g 0 100000 65536
lxc.rootfs = overlayfs:/home/eric/.local/share/lxc/base-ubuntu_14_04_4/rootfs:/home/eric/.local/share/lxc/X11/delta0
lxc.pivotdir = lxc_putold
lxc.cgroup.devices.deny = a
lxc.cgroup.devices.allow = c 4:9 rw
```

tty9:

```
$ ls -hl /dev/tty9
crw--w---- 1 root tty 4, 9 Apr 28 02:43 /dev/tty9
```
# Steps to reproduce
1. Create container
2. Modify container config
3. Start container

**Comment by stgraber** [comment]
Please run "lxc-start -n X11 -o debug -l debug" and attach the content of the "debug" file.

**Comment by naisanza** [comment]
@stgraber Attached!

[debug.txt](https://github.com/lxc/lxc/files/244490/debug.txt):

```
      lxc-start 1462161373.542 INFO     lxc_start_ui - lxc_start.c:main:264 - using rcfile /home/eric/.local/share/lxc/X11/config
      lxc-start 1462161373.543 INFO     lxc_confile - confile.c:config_idmap:1378 - read uid map: type u nsid 0 hostid 100000 range 65536
      lxc-start 1462161373.543 INFO     lxc_confile - confile.c:config_idmap:1378 - read uid map: type g nsid 0 hostid 100000 range 65536
      lxc-start 1462161373.543 WARN     lxc_log - log.c:lxc_log_init:316 - lxc_log_init called with log already initialized
      lxc-start 1462161373.544 WARN     lxc_cgmanager - cgmanager.c:cgm_get:985 - do_cgm_get exited with error
      lxc-start 1462161373.544 INFO     lxc_lsm - lsm/lsm.c:lsm_init:48 - LSM security driver AppArmor
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .reject_force_umount  # comment this to allow umount -f;  not recommended.
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:426 - Adding native rule for reject_force_umount action 0
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:216 - Setting seccomp rule to reject force umounts

      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:429 - Adding compat rule for reject_force_umount action 0
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:do_resolve_add_rule:216 - Setting seccomp rule to reject force umounts

      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .[all].
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .kexec_load errno 1.
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:426 - Adding native rule for kexec_load action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:429 - Adding compat rule for kexec_load action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .open_by_handle_at errno 1.
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:426 - Adding native rule for open_by_handle_at action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:429 - Adding compat rule for open_by_handle_at action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .init_module errno 1.
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:426 - Adding native rule for init_module action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:429 - Adding compat rule for init_module action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .finit_module errno 1.
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:426 - Adding native rule for finit_module action 327681
      lxc-start 1462161373.545 WARN     lxc_seccomp - seccomp.c:do_resolve_add_rule:233 - Seccomp: got negative # for syscall: finit_module
      lxc-start 1462161373.545 WARN     lxc_seccomp - seccomp.c:do_resolve_add_rule:234 - This syscall will NOT be blacklisted
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:429 - Adding compat rule for finit_module action 327681
      lxc-start 1462161373.545 WARN     lxc_seccomp - seccomp.c:do_resolve_add_rule:233 - Seccomp: got negative # for syscall: finit_module
      lxc-start 1462161373.545 WARN     lxc_seccomp - seccomp.c:do_resolve_add_rule:234 - This syscall will NOT be blacklisted
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:324 - processing: .delete_module errno 1.
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:426 - Adding native rule for delete_module action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:429 - Adding compat rule for delete_module action 327681
      lxc-start 1462161373.545 INFO     lxc_seccomp - seccomp.c:parse_config_v2:436 - Merging in the compat seccomp ctx into the main one
      lxc-start 1462161373.546 DEBUG    lxc_conf - conf.c:lxc_create_tty:3791 - allocated pty '/dev/pts/60' (5/6)
      lxc-start 1462161373.546 DEBUG    lxc_conf - conf.c:lxc_create_tty:3791 - allocated pty '/dev/pts/61' (7/8)
      lxc-start 1462161373.546 DEBUG    lxc_conf - conf.c:lxc_create_tty:3791 - allocated pty '/dev/pts/62' (9/10)
      lxc-start 1462161373.546 DEBUG    lxc_conf - conf.c:lxc_create_tty:3791 - allocated pty '/dev/pts/63' (11/12)
      lxc-start 1462161373.546 INFO     lxc_conf - conf.c:lxc_create_tty:3802 - tty's configured
      lxc-start 1462161373.546 DEBUG    lxc_start - start.c:setup_signal_fd:263 - sigchild handler set
      lxc-start 1462161373.546 DEBUG    lxc_console - console.c:lxc_console_peer_default:500 - opening /dev/tty for console peer
      lxc-start 1462161373.546 DEBUG    lxc_console - console.c:lxc_console_peer_default:506 - using '/dev/tty' as console
      lxc-start 1462161373.546 DEBUG    lxc_console - console.c:lxc_console_sigwinch_init:179 - 4296 got SIGWINCH fd 17
      lxc-start 1462161373.546 DEBUG    lxc_console - console.c:lxc_console_winsz:88 - set winsz dstfd:14 cols:141 rows:47
      lxc-start 1462161373.777 INFO     lxc_start - start.c:lxc_init:463 - 'X11' is initialized
      lxc-start 1462161373.777 DEBUG    lxc_start - start.c:__lxc_start:1099 - Not dropping cap_sys_boot or watching utmp
      lxc-start 1462161373.777 INFO     lxc_start - start.c:lxc_spawn:832 - Cloning a new user namespace
      lxc-start 1462161373.777 INFO     lxc_cgroup - cgroup.c:cgroup_init:62 - cgroup driver cgmanager initing for X11
      lxc-start 1462161373.786 INFO     lxc_cgmanager - cgmanager.c:cgm_setup_limits:1389 - cgroup limits have been setup
      lxc-start 1462161373.879 NOTICE   lxc_start - start.c:do_start:685 - switching to gid/uid 0 in new user namespace
      lxc-start 1462161373.887 INFO     bdev - bdev.c:overlayfs_mount:2309 - overlayfs: mounted /home/eric/.local/share/lxc/base-ubuntu_14_04_4/rootfs onto /usr/lib/x86_64-linux-gnu/lxc options upperdir=/home/eric/.local/share/lxc/X11/delta0,lowerdir=/home/eric/.local/share/lxc/base-ubuntu_14_04_4/rootfs
      lxc-start 1462161373.887 DEBUG    lxc_conf - conf.c:setup_rootfs:1625 - mounted 'overlayfs:/home/eric/.local/share/lxc/base-ubuntu_14_04_4/rootfs:/home/eric/.local/share/lxc/X11/delta0' on '/usr/lib/x86_64-linux-gnu/lxc'
      lxc-start 1462161373.887 INFO     lxc_conf - conf.c:setup_utsname:912 - 'X11' hostname has been setup
      lxc-start 1462161373.888 DEBUG    lxc_conf - conf.c:setup_hw_addr:2682 - mac address '00:16:3e:d7:c7:bc' on 'eth0' has been setup
      lxc-start 1462161373.888 DEBUG    lxc_conf - conf.c:setup_netdev:2909 - 'eth0' has been setup
      lxc-start 1462161373.888 INFO     lxc_conf - conf.c:setup_network:2930 - network has been setup
      lxc-start 1462161373.888 DEBUG    lxc_conf - conf.c:check_autodev:4031 - Set exec command to /sbin/init
      lxc-start 1462161373.889 INFO     lxc_conf - conf.c:check_autodev:4069 - Autodev not required.
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted 'proc' on '/usr/lib/x86_64-linux-gnu/lxc/proc', type 'proc'
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted 'sysfs' on '/usr/lib/x86_64-linux-gnu/lxc/sys', type 'sysfs'
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /sys/fs/fuse/connections on /usr/lib/x86_64-linux-gnu/lxc/sys/fs/fuse/connections to respect bind or remount options
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /sys/fs/fuse/connections was 4096, required extra flags are 0
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/sys/fs/fuse/connections' on '/usr/lib/x86_64-linux-gnu/lxc/sys/fs/fuse/connections', type 'none'
      lxc-start 1462161373.889 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /sys/kernel/debug on /usr/lib/x86_64-linux-gnu/lxc/sys/kernel/debug to respect bind or remount options
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /sys/kernel/debug was 4096, required extra flags are 0
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/sys/kernel/debug' on '/usr/lib/x86_64-linux-gnu/lxc/sys/kernel/debug', type 'none'
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /sys/kernel/security on /usr/lib/x86_64-linux-gnu/lxc/sys/kernel/security to respect bind or remount options
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /sys/kernel/security was 4096, required extra flags are 0
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/sys/kernel/security' on '/usr/lib/x86_64-linux-gnu/lxc/sys/kernel/security', type 'none'
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /sys/fs/pstore on /usr/lib/x86_64-linux-gnu/lxc/sys/fs/pstore to respect bind or remount options
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /sys/fs/pstore was 4096, required extra flags are 0
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/sys/fs/pstore' on '/usr/lib/x86_64-linux-gnu/lxc/sys/fs/pstore', type 'none'
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/console on /usr/lib/x86_64-linux-gnu/lxc/dev/console to respect bind or remount options
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/console was 4096, required extra flags are 0
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/console' on '/usr/lib/x86_64-linux-gnu/lxc/dev/console', type 'none'
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/full on /usr/lib/x86_64-linux-gnu/lxc/dev/full to respect bind or remount options
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/full was 4096, required extra flags are 0
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/full' on '/usr/lib/x86_64-linux-gnu/lxc/dev/full', type 'none'
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/null on /usr/lib/x86_64-linux-gnu/lxc/dev/null to respect bind or remount options
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/null was 4096, required extra flags are 0
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.890 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/null' on '/usr/lib/x86_64-linux-gnu/lxc/dev/null', type 'none'
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/random on /usr/lib/x86_64-linux-gnu/lxc/dev/random to respect bind or remount options
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/random was 4096, required extra flags are 0
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/random' on '/usr/lib/x86_64-linux-gnu/lxc/dev/random', type 'none'
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/tty on /usr/lib/x86_64-linux-gnu/lxc/dev/tty to respect bind or remount options
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/tty was 4096, required extra flags are 0
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/tty' on '/usr/lib/x86_64-linux-gnu/lxc/dev/tty', type 'none'
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/urandom on /usr/lib/x86_64-linux-gnu/lxc/dev/urandom to respect bind or remount options
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/urandom was 4096, required extra flags are 0
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/urandom' on '/usr/lib/x86_64-linux-gnu/lxc/dev/urandom', type 'none'
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /dev/zero on /usr/lib/x86_64-linux-gnu/lxc/dev/zero to respect bind or remount options
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /dev/zero was 4096, required extra flags are 0
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2092 - mountflags already was 4096, skipping remount
      lxc-start 1462161373.891 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/dev/zero' on '/usr/lib/x86_64-linux-gnu/lxc/dev/zero', type 'none'
      lxc-start 1462161373.892 ERROR    lxc_utils - utils.c:safe_mount:1284 - No such file or directory - Mount of '/sys/firmware/efi/efivars' onto '/usr/lib/x86_64-linux-gnu/lxc/sys/firmware/efi/efivars' failed
      lxc-start 1462161373.892 INFO     lxc_conf - conf.c:mount_entry:2057 - failed to mount '/sys/firmware/efi/efivars' on '/usr/lib/x86_64-linux-gnu/lxc/sys/firmware/efi/efivars' (optional): No such file or directory
      lxc-start 1462161373.892 DEBUG    lxc_conf - conf.c:mount_entry:2068 - remounting /proc/sys/fs/binfmt_misc on /usr/lib/x86_64-linux-gnu/lxc/proc/sys/fs/binfmt_misc to respect bind or remount options
      lxc-start 1462161373.892 DEBUG    lxc_conf - conf.c:mount_entry:2083 - (at remount) flags for /proc/sys/fs/binfmt_misc was 4110, required extra flags are 14
      lxc-start 1462161373.892 DEBUG    lxc_conf - conf.c:mount_entry:2118 - mounted '/proc/sys/fs/binfmt_misc' on '/usr/lib/x86_64-linux-gnu/lxc/proc/sys/fs/binfmt_misc', type 'none'
      lxc-start 1462161373.892 INFO     lxc_conf - conf.c:mount_file_entries:2480 - mount points have been setup
      lxc-start 1462161373.892 INFO     lxc_conf - conf.c:setup_dev_console:1848 - console has been setup
      lxc-start 1462161373.893 INFO     lxc_conf - conf.c:setup_tty:1041 - 4 tty(s) has been setup
      lxc-start 1462161373.893 INFO     lxc_conf - conf.c:do_tmp_proc_mount:4095 - I am 1, /proc/self points to '1'
      lxc-start 1462161373.893 DEBUG    lxc_conf - conf.c:setup_rootfs_pivot_root:1211 - created '/usr/lib/x86_64-linux-gnu/lxc/lxc_putold' directory
      lxc-start 1462161373.893 DEBUG    lxc_conf - conf.c:setup_rootfs_pivot_root:1214 - mountpoint for old rootfs is '/usr/lib/x86_64-linux-gnu/lxc/lxc_putold'
      lxc-start 1462161373.894 DEBUG    lxc_conf - conf.c:setup_rootfs_pivot_root:1227 - pivot_root syscall to '/usr/lib/x86_64-linux-gnu/lxc' successful
      lxc-start 1462161373.904 INFO     lxc_conf - conf.c:umount_oldrootfs:1165 - lazy unmount of '/lxc_putold'
      lxc-start 1462161373.904 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/dev'
      lxc-start 1462161373.904 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/dev/pts'
      lxc-start 1462161373.904 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/run'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/run/lock'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/run/shm'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/run/user'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys/fs/cgroup'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys/fs/cgroup/systemd'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys/fs/fuse/connections'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys/kernel/debug'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys/kernel/security'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/sys/fs/pstore'
      lxc-start 1462161373.905 WARN     lxc_conf - conf.c:umount_oldrootfs:1175 - failed to unmount '/lxc_putold/proc'
      lxc-start 1462161373.905 INFO     lxc_conf - conf.c:setup_personality:1803 - set personality to '0x0'
      lxc-start 1462161373.905 DEBUG    lxc_conf - conf.c:setup_caps:2596 - drop capability 'sys_module' (16)
      lxc-start 1462161373.905 DEBUG    lxc_conf - conf.c:setup_caps:2596 - drop capability 'mac_admin' (33)
      lxc-start 1462161373.905 DEBUG    lxc_conf - conf.c:setup_caps:2596 - drop capability 'mac_override' (32)
      lxc-start 1462161373.905 DEBUG    lxc_conf - conf.c:setup_caps:2596 - drop capability 'sys_time' (25)
      lxc-start 1462161373.905 DEBUG    lxc_conf - conf.c:setup_caps:2605 - capabilities have been setup
      lxc-start 1462161373.905 NOTICE   lxc_conf - conf.c:lxc_setup:4370 - 'X11' is setup.
      lxc-start 1462161373.906 DEBUG    lxc_cgmanager - cgmanager.c:cgm_setup_limits:1385 - cgroup 'devices.deny' set to 'a'
      lxc-start 1462161373.907 DEBUG    lxc_cgmanager - cgmanager.c:cgm_setup_limits:1385 - cgroup 'devices.allow' set to 'c 4:9 rw'
      lxc-start 1462161373.907 INFO     lxc_cgmanager - cgmanager.c:cgm_setup_limits:1389 - cgroup limits have been setup
      lxc-start 1462161373.907 INFO     lxc_apparmor - lsm/apparmor.c:apparmor_process_label_set:155 - changed apparmor profile to lxc-container-default
      lxc-start 1462161373.907 NOTICE   lxc_start - start.c:start:1200 - exec'ing '/sbin/init'
      lxc-start 1462161373.908 NOTICE   lxc_start - start.c:post_start:1211 - '/sbin/init' started with pid '4400'
      lxc-start 1462161373.909 WARN     lxc_start - start.c:signal_handler:311 - invalid pid for SIGCHLD
      lxc-start 1462161373.909 DEBUG    lxc_commands - commands.c:lxc_cmd_handler:893 - peer has disconnected
      lxc-start 1462161373.910 DEBUG    lxc_commands - commands.c:lxc_cmd_handler:893 - peer has disconnected
      lxc-start 1462161373.911 DEBUG    lxc_commands - commands.c:lxc_cmd_handler:893 - peer has disconnected
      lxc-start 1462161373.912 DEBUG    lxc_start - start.c:signal_handler:315 - container init process exited
      lxc-start 1462161373.912 DEBUG    lxc_start - start.c:__lxc_start:1164 - Pushing physical nics back to host namespace
      lxc-start 1462161373.912 DEBUG    lxc_start - start.c:__lxc_start:1167 - Tearing down virtual network devices used by container
      lxc-start 1462161373.912 INFO     lxc_error - error.c:lxc_error_set_and_log:49 - child <4400> ended on error (1)
      lxc-start 1462161373.940 ERROR    lxc_start_ui - lxc_start.c:main:341 - The container failed to start.
      lxc-start 1462161373.941 ERROR    lxc_start_ui - lxc_start.c:main:345 - Additional information can be obtained by setting the --logfile and --logpriority options.
```

**Comment by stgraber** [comment]
So the output above indicates that the following was applied:
 - devices.deny = a
 - devices.allow = c 4:9 rw

That means every other device access is being denied, which includes access to useful things like /dev/console, /dev/null, /dev/zero, ... which causes your init process to fail immediately as reported in the log (ended on error (1)).

**Comment by naisanza** [comment]
@stgraber This is my first time configuring cgroups, and stabbing around in the dark, and probably doing it all wrong. 

My goal is to use LXC like terminal services, with each container with its own windows manager, or running a full Desktop Environment

I've only ever used SmartOS, KVM, ESXi, VirtualBox, and VMware but ran everything from the shell; this is my first time trying to get a Desktop Environment running in a container-style hypervisor

Only problem (since I haven't gotten past this part) is getting X server to start, since the first thing I grabbed to try is VNC and it requires a running X server (I believe).

The first error I ran into says it `couldn't open /dev/tty0`

So I just symlinked it from `/dev/tty1`

The next error says it `Cannot open virtual console 7 (No such file or directory)`

So I started looking around and came upon cgroups, and read up until I learned about what the numbers represented in `/proc/devices`

And now, this is the furthest that I am at so far. From everything I've read online, it seems that having a DE is more of a pipedream and an afterthought; and Unity8 is the only "solution" 

But they all require a lot of hands-on config changes, and it's not at the point of just doing `lxc-create -t ubuntu-gnome-desktop -n Dekstop01` and then just pointing your client at the container IP and you have a remote desktop running.

So that's my goal that I'm hoping to get to
